diff --git a/v1/Foo.cpp b/v1/Foo.cpp
index c18decc..4361aee 100644
--- a/v1/Foo.cpp
+++ b/v1/Foo.cpp
@@ -5,7 +5,22 @@ namespace v1 {
   Foo::Foo(int _n) : l(*this, _n, 0, 1) {
     n = _n;
 
-    Gecode::branch(*this, l, Gecode::INT_VAR_DEGREE_MAX(), Gecode::INT_VAL_MIN());
+    // Gecode::branch(*this, l, Gecode::INT_VAR_DEGREE_MAX(), Gecode::INT_VAL_MIN());
+
+    // Gecode::Rnd r(2015U);
+    // Gecode::branch(*this, l, tiebreak(Gecode::INT_VAR_AFC_MAX(1),
+    //                                   Gecode::INT_VAR_DEGREE_MAX()),
+    //                Gecode::INT_VAL_MAX());
+
+    // Gecode::branch(*this, l, tiebreak(Gecode::INT_VAR_DEGREE_MAX(),
+    //                                     Gecode::INT_VAR_AFC_MAX(1)),
+    //                Gecode::INT_VAL_MIN());
+    
+    // Gecode::Rnd r(2015U);
+    // Gecode::branch(*this, l,
+    //                tiebreak(Gecode::INT_VAR_DEGREE_MAX(),
+    //                         Gecode::INT_VAR_RND(r)),
+    //                Gecode::INT_VAL_MIN());
   }
 
   Foo::Foo(bool share, Foo& s) : Gecode::Space(share, s) {
